plan-research/project-minimization
# JBRes-2628 Preparation for testing #2: Settings Upgrade

## Description

The main changes:
1. Rewrote UI for the settings using DSL
2. Split UI logic into several classes
3. Updated stages' settings: now it's possible to configure them flexibly
4. Added a dialog window before minimization to ask about gradle tasks explicitly
5. Fixed several bugs

## How to test

### Manual tests

1. Run `runIde`
2.1. Open the plugin's settings
2.2. Run the minimize action -> see the dialog

## Self-check list

- [x] PR **title** and **description** are clear and aligned with a format.
- [x] **Manual tests** have been provided _optionally_.
- [x] The **documentation** for the functionality I've been working on is up-to-date/provided.
- [x] The **link** to this PR is commented on in the corresponding **YT ticket**.

Hint: [x] is a marked item
